dev/com.ibm.ws.microprofile.metrics.3.0/src/com/ibm/ws/microprofile/metrics30/impl/MetricRegistry30Impl.java

```diff
@@ -0,0 +1,1090 @@
+        return concurrentGauge(metricID.getName(), metricID.getTagsAsArray());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConcurrentGauge concurrentGauge(Metadata inputMetadata) {
+        Metadata metadata = null;
+        if (inputMetadata.getTypeRaw() == MetricType.INVALID) {
+            metadata = Metadata.builder(inputMetadata).withType(MetricType.CONCURRENT_GAUGE).build();
+        } else {
+            metadata = inputMetadata;
+        }
+        return this.concurrentGauge(metadata, null);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConcurrentGauge concurrentGauge(Metadata metadata, Tag... tags) {
+        return getOrAdd(metadata, MetricBuilder30.CONCURRENT_GAUGE, tags);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public SortedMap<MetricID, ConcurrentGauge> getConcurrentGauges() {
+        return getConcurrentGauges(MetricFilter.ALL);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public SortedMap<MetricID, ConcurrentGauge> getConcurrentGauges(MetricFilter filter) {
+        return getMetrics(ConcurrentGauge.class, filter);
+    }
+
+    private ClassLoader getThreadContextClassLoader() {
+        if (usingJava2Security) {
+            return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () -> {
+                return Thread.currentThread().getContextClassLoader();
+            });
+        }
+        return Thread.currentThread().getContextClassLoader();
+    }
+
+    @Override
+    public SimpleTimer simpleTimer(String name) {
+        return this.simpleTimer(name, null);
+    }
+
+    @Override
+    public SimpleTimer simpleTimer(String name, Tag... tags) {
+        Metadata metadata = Metadata.builder().withName(name).withType(MetricType.SIMPLE_TIMER).build();
+
+        if (metadataMID.keySet().contains(name)) {
+            metadata = metadataMID.get(name);
+
+            if (!metadata.getTypeRaw().equals(MetricType.SIMPLE_TIMER)) {
+                throw new IllegalArgumentException(name + " is already used for a different type of metric");
+            }
+        }
+        return this.simpleTimer(metadata, tags);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public SimpleTimer simpleTimer(MetricID metricID) {
+        return simpleTimer(metricID.getName(), metricID.getTagsAsArray());
+    }
+
+    @Override
+    public SimpleTimer simpleTimer(Metadata inputMetadata) {
+        Metadata metadata = null;
+        if (inputMetadata.getTypeRaw() == MetricType.INVALID) {
+            metadata = Metadata.builder(inputMetadata).withType(MetricType.SIMPLE_TIMER).build();
+        } else {
+            metadata = inputMetadata;
+        }
+        return this.simpleTimer(metadata, null);
+    }
+
+    @Override
+    public SimpleTimer simpleTimer(Metadata metadata, Tag... tags) {
+        return getOrAdd(metadata, MetricBuilder30.SIMPLE_TIMER, tags);
+    }
+
+    @Override
+    public SortedMap<MetricID, SimpleTimer> getSimpleTimers() {
```

Comment:
All new methods below. More overloaded methods to register/retreive metrics. They end up calling the other existing overloaded methods.